eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubSnpNoHgvsToDbsnpVariantsWrapperProcessorTest.java

```diff
@@ -60,31 +81,87 @@ public static void tearDownClass() throws Exception {
-                                                     Orientation.FORWARD, Orientation.FORWARD, false, false, false,
-                                                     false, CREATED_DATE, CREATED_DATE, TAXONOMY);
+        SubSnpNoHgvs subSnpNoHgvs = new SubSnpNoHgvs(25928972L, 14718243L, "A", "C", ASSEMBLY, BATCH_HANDLE, BATCH_NAME,
+                                                     CHROMOSOME, CHROMOSOME_START, CONTIG_NAME, CONTIG_START,
+                                                     DbsnpVariantType.SNV, Orientation.FORWARD, Orientation.FORWARD,
+                                                     Orientation.FORWARD, false, false, false, false, CREATED_DATE,
+                                                     CREATED_DATE, TAXONOMY);
 
         DbsnpVariantsWrapper dbsnpVariantsWrapper = processor.process(subSnpNoHgvs);
-        assertSubmittedVariantDeclustered(dbsnpVariantsWrapper);
-        assertDeclusterOperationsRegistered(subSnpNoHgvs, dbsnpVariantsWrapper);
+
+        //Check Decluster
+        assertEquals(1, dbsnpVariantsWrapper.getSubmittedVariants().size());
+        assertNull(dbsnpVariantsWrapper.getSubmittedVariants().get(0).getClusteredVariantAccession());
+
+        //Check Operations
+        assertOperations(dbsnpVariantsWrapper, DECLUSTERED_ALLELES_MISMATCH, subSnpNoHgvs.getSsId(), 0);
+
+        //Check Rs
         assertEquals(subSnpNoHgvs.getRsId(), dbsnpVariantsWrapper.getClusteredVariant().getAccession());
     }
 
-    private void assertSubmittedVariantDeclustered(DbsnpVariantsWrapper dbsnpVariantsWrapper) {
-        assertEquals(1, dbsnpVariantsWrapper.getSubmittedVariants().size());
+    @Test
+    public void processSubmittedVariantTypeMismatch() throws Exception {
+        SubSnpNoHgvs subSnpNoHgvsTypeMismatch = new SubSnpNoHgvs(25928972L, 14718243L, "G", "G/-/T/TT", ASSEMBLY,
+                                                                 BATCH_HANDLE, BATCH_NAME, CHROMOSOME, CHROMOSOME_START,
+                                                                 CONTIG_NAME, CONTIG_START, DbsnpVariantType.DIV,
+                                                                 Orientation.FORWARD, Orientation.FORWARD,
+                                                                 Orientation.FORWARD, false, false, false, false,
+                                                                 CREATED_DATE, CREATED_DATE, TAXONOMY);
+
+        DbsnpVariantsWrapper dbsnpVariantsWrapper = processor.process(subSnpNoHgvsTypeMismatch);
+
+        //Check Decluster
+        assertEquals(3, dbsnpVariantsWrapper.getSubmittedVariants().size());
+        assertNotNull(dbsnpVariantsWrapper.getSubmittedVariants().get(0).getClusteredVariantAccession());
+        assertNull(dbsnpVariantsWrapper.getSubmittedVariants().get(1).getClusteredVariantAccession());
+        assertNull(dbsnpVariantsWrapper.getSubmittedVariants().get(2).getClusteredVariantAccession());
```

Comment:
I don't understand which is the order of the alternate alleles here but as @pabarcgar said, it doesn't look like they should be declustered this way.

Reply:
G is the reference so it is not counted as variant, so we have -/T/TT as alternates. G>'' (DEL) would be the first one and this will determine the RS EVA type. The other variants will be declustered (SNV, INS)